jans-auth-server/common/src/test/java/io/jans/as/common/RedirectUriTest.java

```diff
@@ -0,0 +1,342 @@
+/*
+ * Janssen Project software is available under the Apache License (2004). See http://www.apache.org/licenses/ for full text.
+ *
+ * Copyright (c) 2020, Janssen Project
+ */
+
+package io.jans.as.common;
+
+import io.jans.as.common.util.RedirectUri;
+import io.jans.as.model.authorize.AuthorizeResponseParam;
+import io.jans.as.model.common.ResponseMode;
+import io.jans.as.model.common.ResponseType;
+import io.jans.as.model.crypto.AbstractCryptoProvider;
+import io.jans.as.model.crypto.encryption.BlockEncryptionAlgorithm;
+import io.jans.as.model.crypto.encryption.KeyEncryptionAlgorithm;
+import io.jans.as.model.crypto.signature.SignatureAlgorithm;
+import io.jans.as.model.exception.CryptoProviderException;
+import io.jans.as.model.exception.InvalidJweException;
+import io.jans.as.model.jwe.Jwe;
+import io.jans.as.model.util.Util;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.testng.MockitoTestNGListener;
+import org.testng.annotations.Listeners;
+import org.testng.annotations.Test;
+
+import java.io.UnsupportedEncodingException;
+import java.math.BigInteger;
+import java.net.URLEncoder;
+import java.security.interfaces.RSAPublicKey;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+import static io.jans.as.model.authorize.AuthorizeRequestParam.*;
+import static io.jans.as.model.authorize.AuthorizeResponseParam.RESPONSE;
+import static org.mockito.ArgumentMatchers.anyObject;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.*;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+@Listeners(MockitoTestNGListener.class)
+public class RedirectUriTest {
+
+    @Mock
+    private AbstractCryptoProvider cryptoProvider;
+
+    @Test
+    public void parseQueryString_queryStringNull_EmptyResult() throws CryptoProviderException {
+        RedirectUri redirectUri = getRedirectUriTemplateGetQueryString(null, null, null, null);
+        redirectUri.parseQueryString(null);
+        assertEquals(redirectUri.getResponseParamentersSize(), 0);
+    }
+
+    @Test
+    public void parseQueryString_oneParam_size1() throws CryptoProviderException {
+        RedirectUri redirectUri = getRedirectUriTemplateGetQueryString(null, null, null, null);
+        redirectUri.parseQueryString(AuthorizeResponseParam.EXPIRES_IN + "=");
+        assertEquals(redirectUri.getResponseParamentersSize(), 1);
```

Comment:
I'd suggest to add more asserts here in order to validate right params were gotten.

Reply:
More Asserts, was added